server/src/main/java/org/opensearch/plugins/PluginSecurity.java

```diff
@@ -171,4 +224,12 @@ public static Set<String> parsePermissions(Path file, Path tmpDir) throws IOExce
         }
         return Collections.list(actualPermissions.elements()).stream().map(PluginSecurity::formatPermission).collect(Collectors.toSet());
     }
+
+    /**
+     * Parses plugin-permissions.yml file.
+     */
+    @SuppressWarnings("removal")
+    public static Settings parseRequestedActions(Path file) throws IOException {
+        return Settings.builder().loadFromPath(file).build();
```

Comment:
I chose to use the Settings object in here since Settings already supports reading in Yaml files.